### ISSUE TITLE
Nullability mismatch is not repoted for collection expressions' implicit `.Add` call receivers

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#79803]
**Version Used**: 2e23a0df5bf8d49f6bf856287b0eb10de08f97d9

**Steps to Reproduce**:

Compile and run the following code:
```
using System;
using System.Collections;
using System.Collections.Generic;

string? obj = null;
C<string> c = [obj];
foreach (string o in c)
{
  Console.WriteLine(o.Length);
}

// both lilnes below would report CS8620:
// c.Add(obj);
// c = new() { obj };
                         
class C<T> : IEnumerable<T>
{
  internal List<T> _list = new();
                         
  IEnumerator IEnumerable.GetEnumerator() => _list.GetEnumerator();
  public IEnumerator<T> GetEnumerator() => _list.GetEnumerator();
}
                         
static class Ext
{
  public static void Add<T>(this C<T> c, T t) { c._list.Add(t); }
}
```

**Diagnostic Id**:

`CS8620 Argument type cannot be used for a parameter because of differences in the nullability of reference types`
*
*Expected Behavior**:
CS8620 warning reported for `[obj]` collection expression.
Note that if collection expression is replaced with either collection initializer or explicit `.Add` call the warning is correctly reported

**Actual Behavior**:
No warnings at all. The code crashes at runtime with a `NullReferenceException`

**Comment by RikkiGibson** [comment]
I expected #78926 to solve this but it looks like the compiler in this report is newer than that. I'll try to take a look soon
